src/visual.ts

```diff
@@ -104,6 +110,26 @@ module powerbi.extensibility.visual {
             const objectCheck = metadata && metadata.objects;
             const properties = DataViewObjects.getObject(objectCheck, "general") as any || {}; 
             let searchText = "";
+            let dataView: DataView = options.dataViews[0];
+
+            this.visualSettings = VisualSettings.parse<VisualSettings>(dataView);
+            this.visualSettings.search.height = Math.max(24, this.visualSettings.search.height);
+
+            //search box input style
+            this.searchBox.placeholder = this.visualSettings.search.placeholder;
+            this.searchBox.style.fontSize = this.visualSettings.search.fontSize.toString()+'px';
+            this.searchBox.style.height = this.visualSettings.search.height.toString()+'px';
+            this.searchBox.style.paddingRight = (2*(this.visualSettings.search.height+3)).toString()+'px';
+            this.searchBox.style.borderColor = this.visualSettings.search.outline;
+
+            //clear button style
+            this.clearButton.style.width = (this.visualSettings.search.height-4).toString()+'px';
+            this.clearButton.style.height = (this.visualSettings.search.height-4).toString()+'px';
+
+            //search button style
+            this.searchButton.style.width = (this.visualSettings.search.height-4).toString()+'px';
+            this.searchButton.style.height = (this.visualSettings.search.height-4).toString()+'px';
+            this.searchButton.style.right = (this.visualSettings.search.height - 4 + 3 + 3).toString() + 'px';
```

Comment:
We try to avoid using numbers in code without explanation of their origin. Could you, please, consider moving the numbers here and in lines above into variables with meaningful names?